pandas/core/series.py

```diff
@@ -354,15 +354,16 @@ def _init_dict(self, data, index=None, dtype=None):
         # Looking for NaN in dict doesn't work ({np.nan : 1}[float('nan')]
         # raises KeyError), so we iterate the entire dict, and align
         if data:
-            keys, values = zip(*data.items())
-            values = list(values)
+            # GH:34717
+            keys = tuple(data.keys())
+            values = list(data.values())  # Generating list of values- faster way
```

Comment:
for future readers can you clarify what this is faster than?

Reply:
@jbrockmendel  i had added it in the past. However the reviewers suggested just to add on the GH issue number therefore i had to remove the additional comments for more clarity about how and why.

Reply:
@Rohith295 since we have a specific perf fix we need document this slightly more here.

Reply:
@jreback alright, then i will add some comments.